components/director/cmd/director/main.go

```diff
@@ -468,3 +506,51 @@ func webhookService() webhook.WebhookService {
 
 	return webhook.NewService(webhookRepo, uidSvc)
 }
+
+func getAsyncDirective(ctx context.Context, cfg config, transact persistence.Transactioner, appRepo application.ApplicationRepository) func(context.Context, interface{}, gqlgen.Resolver, graphql.OperationType, *graphql.WebhookType, *string) (res interface{}, err error) {
+	resourceFetcherFunc := func(ctx context.Context, tenantID, resourceID string) (model.Entity, error) {
+		return appRepo.GetByID(ctx, tenantID, resourceID)
+	}
+
+	scheduler, err := buildScheduler(ctx, cfg)
+	exitOnError(err, "Error while creating operations scheduler")
+
+	return operation.NewDirective(transact, webhookService().List, resourceFetcherFunc, tenant.LoadFromContext, scheduler).HandleOperation
+}
+
+func buildScheduler(ctx context.Context, config config) (operation.Scheduler, error) {
+	if config.DisableAsyncMode {
+		log.C(ctx).Info("Async operations are disabled")
+		return &operation.DisabledScheduler{}, nil
+	}
+
+	cfg, err := getKubeConfig(ctx, config)
+	exitOnError(err, "Failed to get cluster config for operations k8s client")
+
+	cfg.Timeout = config.ClientTimeout
+	k8sClient, err := client.NewForConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+	operationsK8sClient := k8sClient.Operations(config.OperationsNamespace)
+
+	return k8s.NewScheduler(operationsK8sClient), nil
+}
+
+func getKubeConfig(ctx context.Context, config config) (*rest.Config, error) {
+	if !config.OutsideCluster {
+		log.C(ctx).Info("Using in-cluster configuration for k8s client")
+		return rest.InClusterConfig()
+	}
+
+	var kubeconfig *string
+	home := homedir.HomeDir()
+	if home == "" {
+		return nil, errors.New("failed to get home dir in order to build k8s client for scheduler")
+	}
+
+	log.C(ctx).Info("Using local configuration for k8s client")
+	kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	flag.Parse()
```

Comment:
this flag is already defined by controller-runtime, also the outsideCluster variable seems unneeded. If config lookup is done in the correct order, it should seemlessly look for inclusterconfig if it is not found else where

I'd propose to drop this method and instead use `func GetConfig() (*rest.Config, error)`

from the godoc 
```
// Config precedence
//
// * --kubeconfig flag pointing at a file
//
// * KUBECONFIG environment variable pointing at a file
//
// * In-cluster config if running in cluster
//
// * $HOME/.kube/config if exists
```